### PR TITLE
bench-4152:  implement filterableGroupFilterBuilder

### PR DESCRIPTION
**Pull request by dexamundsen**
Local testing sample

Select all group gene

```
SELECT COUNT(id) AS T_CTDT, gender 
FROM `verily-tanagra-dev.aou_test_data_SC2023Q3R2_index_090424`.T_ENT_person 
WHERE (id IN (SELECT flattened_person_id 
              FROM (SELECT v.variant_row_num, flattened_person_id
                    /* Wrap variant_to_person table in a SELECT DISTINCT because there is a duplicate row in the test data. */
                    FROM (SELECT DISTINCT vid, person_ids 
                          FROM `verily-tanagra-dev.aou_test_data_SC2023Q3R2.variant_to_person`) AS vtop
                    JOIN (SELECT ROW_NUMBER() OVER (ORDER BY vid) AS variant_row_num, vid 
                          FROM `verily-tanagra-dev.aou_test_data_SC2023Q3R2.prep_vat`) AS v 
                    ON v.vid = vtop.vid
                    CROSS JOIN UNNEST(vtop.person_ids) AS flattened_person_id)
              WHERE variant_row_num IN (SELECT id 
                                        FROM `verily-tanagra-dev.aou_test_data_SC2023Q3R2_index_090424`.T_ENT_variant 
                                        WHERE (REGEXP_CONTAINS(UPPER(gene), UPPER('gene'))) 
                                          AND (EXISTS (SELECT * 
                                                      FROM UNNEST(consequence) AS flattened 
                                                      WHERE flattened IN ('5_prime_UTR_variant')))
                                          AND (id NOT IN (11983,25881,32049,12637)))))
  OR (id IN (SELECT flattened_person_id 
            FROM (SELECT v.variant_row_num, flattened_person_id
                  /* Wrap variant_to_person table in a SELECT DISTINCT because there is a duplicate row in the test data. */
                  FROM (SELECT DISTINCT vid, person_ids 
                        FROM `verily-tanagra-dev.aou_test_data_SC2023Q3R2.variant_to_person`) AS vtop
                  JOIN (SELECT ROW_NUMBER() OVER (ORDER BY vid) AS variant_row_num, vid 
                        FROM `verily-tanagra-dev.aou_test_data_SC2023Q3R2.prep_vat`) AS v
                  ON v.vid = vtop.vid
                  CROSS JOIN UNNEST(vtop.person_ids) AS flattened_person_id) 
                  WHERE variant_row_num IN (1361,1350,1358,1348,1363))) 
GROUP BY gender ORDER BY T_CTDT DESC LIMIT 1000000
```

### CODE REVIEW

File: service/src/main/java/bio/terra/tanagra/app/authentication/AuthInterceptor.java
Comment (dexamundsen) [inline service/src/main/java/bio/terra/tanagra/app/authentication/AuthInterceptor.java:74]: temp comments, will be removed

File: underlay/src/main/proto/criteriaselector/dataschema/filterable_group.proto
Comment (dexamundsen) [inline underlay/src/main/proto/criteriaselector/dataschema/filterable_group.proto:26]: Could we change query to ValueData instead?

File: underlay/src/main/java/bio/terra/tanagra/filterbuilder/impl/core/EntityGroupFilterBuilderBase.java
Comment (dexamundsen) [inline underlay/src/main/java/bio/terra/tanagra/filterbuilder/impl/core/EntityGroupFilterBuilderBase.java:391]: moved to EntityGroupFilterUtils.buildGroupItemsFilterFromIds

File: underlay/src/main/proto/criteriaselector/dataschema/filterable_group.proto
Comment (dexamundsen) [inline underlay/src/main/proto/criteriaselector/dataschema/filterable_group.proto:29]: renamed to keep it consistent with other schema

File: underlay/src/main/java/bio/terra/tanagra/filterbuilder/impl/core/utils/EntityGroupFilterUtils.java
Comment (tjennison-work) [inline underlay/src/main/java/bio/terra/tanagra/filterbuilder/impl/core/utils/EntityGroupFilterUtils.java:115]: nit: This function and the next have a mix of "not" and "non" in both code and comments. I like "non" better but there are currently more "not".

**Review by tjennison-work — COMMENTED** [review]
LGTM pending tests.

File: underlay/src/main/java/bio/terra/tanagra/filterbuilder/impl/core/EntityGroupFilterBuilderBase.java
Comment (dexamundsen) [inline underlay/src/main/java/bio/terra/tanagra/filterbuilder/impl/core/EntityGroupFilterBuilderBase.java:281]: remove unused param

File: underlay/src/main/java/bio/terra/tanagra/filterbuilder/impl/core/utils/EntityGroupFilterUtils.java
Comment (dexamundsen) [reply underlay/src/main/java/bio/terra/tanagra/filterbuilder/impl/core/utils/EntityGroupFilterUtils.java:114]: sgtm, Renames all notPrimaryEntity to nonPrimaryEntity

File: underlay/src/main/java/bio/terra/tanagra/filterbuilder/impl/core/FilterableGroupFilterBuilder.java
Comment (tjennison-work) [inline underlay/src/main/java/bio/terra/tanagra/filterbuilder/impl/core/FilterableGroupFilterBuilder.java:49]: Returning `null` here makes an empty selection do nothing. What it should actually do is filter for participants who have any variant data. I think adding a `|| filtersToOr.isEmpty()` on line 89 might be enough.

**Review by tjennison-work — APPROVED** [review]
Were you still going to add tests to this PR?

**Comment by dexamundsen** [comment]
> Were you still going to add tests to this PR?

Yes, will add tests before this is merged. Put it out so that you have these changes for UI is needed / Vumc has updates.

File: underlay/src/main/java/bio/terra/tanagra/filterbuilder/impl/core/FilterableGroupFilterBuilder.java
Comment (dexamundsen) [reply underlay/src/main/java/bio/terra/tanagra/filterbuilder/impl/core/FilterableGroupFilterBuilder.java:49]: sg, done

**Pull request merged**
